### PR TITLE
Register placement manipulator with lua

### PR DESCRIPTION
**Pull request by adam4813**
Send onKeyUp event to lua from game.cpp.

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: client/game.cpp [hunk 1]
```diff
@@ -217,19 +223,7 @@ void Game::ProcessEvents() {
...
-		case GLFW_KEY_4: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[3])); break;
-		case GLFW_KEY_5: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[4])); break;
-		case GLFW_KEY_6: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[5])); break;
-		case GLFW_KEY_7: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[6])); break;
-		case GLFW_KEY_8: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[7])); break;
-		case GLFW_KEY_9: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[8])); break;
-		case GLFW_KEY_0: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[9])); break;
-		default: break;
-		}
+		this->lua_sys.CallFunctions("onKeyUp", data->key);
```
Comment (Meisaka) [inline client/game.cpp:226]: :+1:

**Review by Meisaka — APPROVED** [review]
I have some issues with the assets update:
- ` scripts/key_codes.lua` and `scripts/test.lua` are missing a newline at EOF
- the `scripts/schematics/*.lua` scripts are okay, but their contents is mostly data, which would be better suited as json and then loaded into either Lua or some engine struct
- `build-blocks/texture.png` and `build-blocks/texture_normal.png` are a irregular size (800x800) all textures should be a power of 2 to function well in the graphics pipeline, the color texture shows signs of jpeg artifacts which degrades quality and increases PNG file size, the normal-map has vertical bias (was that intended?), the textures are also too large for the amount of information they contain, the textures are redundant along the Y axis and could be reduced.
- but I'm not here to critique art/assests PRs, now am I? :waffle:

The code in this PR is good though.

**Comment by adam4813** [comment]
I never thought to load JSON into lua...good idea. Any suggestions in how to best do that?

**Comment by Meisaka** [comment]
there are probably libraries written in Lua to parse json to Lua tables (easier?), or you could make the engine do it (faster?).

File: client/game.cpp (hunk 1 above)
Comment (blexaer) [reply client/game.cpp:226]: that's really nice

**Review by blexaer — APPROVED** [review]
Problems with assets as Meisaka pointed out which should be solved
Other than that I got nothing, code looks good.

**Pull request merged**